src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -23,6 +23,84 @@ public static class BufferExtensions
             return array.Items.Slice(0, array.Count);
         }
 
+        /// <summary>
+        /// Creates a new readonly span over the portion of the target string.
+        /// </summary>
+        /// <param name="text">The target string.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is null.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe ReadOnlySpan<char> Slice(this string text)
+        {
+            if (text == null)
+                throw new ArgumentNullException(nameof(text));
+
+            int textLength = text.Length;
+
+            if (textLength == 0) return ReadOnlySpan<char>.Empty;
+
+            fixed (char* charPointer = text)
+            {
+                return ReadOnlySpan<char>.DangerousCreate(text, ref *charPointer, textLength);
+            }
+        }
+
+        /// <summary>
+        /// Creates a new readonly span over the portion of the target string, beginning at 'start'.
+        /// </summary>
+        /// <param name="text">The target string.</param>
+        /// <param name="start">The index at which to begin this slice.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is null.</exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="start"/> index is not in range (&lt;0 or &gt;Length).
+        /// </exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe ReadOnlySpan<char> Slice(this string text, int start)
+        {
+            if (text == null)
+                throw new ArgumentNullException(nameof(text));
+
+            int textLength = text.Length;
+
+            if ((uint) start > (uint) textLength)
+                throw new ArgumentOutOfRangeException(nameof(start));
+
+            if (textLength == 0) return ReadOnlySpan<char>.Empty;
```

Comment:
Yes. I would imagine this would be for performance reasons and not for correctness, right?
I ask this because if (textLength - start) == 0, we just pass in 0 length to the DangerousCreate method.